### PR TITLE
Start of community reward

### PR DESCRIPTION
**Pull request by trianglesphere**
### Description

This PR renames payments to rewards, and gets the community reward amount as a third return value from a governance contract.

### Tested

Tested in e2e tests.

### Other changes

This PR requires coordination with https://github.com/celo-org/celo-monorepo/pull/2328 in the monorepo

### Related issues

- Fixes #[issue number here]

### Backwards compatibility

Nope.

### CODE REVIEW

File: consensus/istanbul/backend/pos.go [hunk 1]
```diff
@@ -37,22 +37,23 @@ import (
...
 		return err
 	}
-	validatorEpochPayment, totalVoterRewards, err := epoch_rewards.CalculateTargetEpochPaymentAndRewards(header, state)
+	// TODO(joshua): fix CalculateTargetEpochRewards to properly calc communityRewards
+	validatorReward, totalVoterRewards, communityReward, err := epoch_rewards.CalculateTargetEpochRewards(header, state)
 	if err != nil {
 		return err
 	}
-	log.Debug("Calculated target epoch payment and rewards", "validatorEpochPayment", validatorEpochPayment, "totalVoterRewards", totalVoterRewards)
+	log.Debug("Calculated target rewards", "validatorReward", validatorReward, "totalVoterRewards", totalVoterRewards)
```
Comment (asaj) [inline consensus/istanbul/backend/pos.go]: Log community reward?

File: consensus/istanbul/backend/pos.go [hunk 2]
```diff
@@ -62,12 +63,17 @@ func (sb *Backend) distributeEpochPaymentsAndRewards(header *types.Header, state
 		return err
 	}
 
-	totalEpochPayments, err := sb.distributeEpochPayments(header, state, valSet, validatorEpochPayment)
+	totalValidatorRewards, err := sb.distributeValidatorRewards(header, state, valSet, validatorReward)
+	if err != nil {
+		return err
+	}
+	// TODO(joshua): properly distribute community rewards here
```
Comment (asaj) [inline consensus/istanbul/backend/pos.go]: What is missing?

File: consensus/istanbul/backend/pos.go [hunk 3]
```diff
@@ -133,43 +141,47 @@ func (sb *Backend) updateValidatorScores(header *types.Header, state *state.Stat
...
-func (sb *Backend) distributeEpochRewards(header *types.Header, state *state.StateDB, valSet []istanbul.Validator, maxTotalRewards *big.Int, uptimes []*big.Int) (*big.Int, error) {
-	totalEpochRewards := big.NewInt(0)
-
-	// Fixed epoch reward to the community fund.
-	// TODO(asa): This should be a fraction of the overall reward to stakers.
-	communityEpochReward := big.NewInt(params.Ether)
+func (sb *Backend) distributeCommunityRewards(header *types.Header, state *state.StateDB, communityReward *big.Int) (*big.Int, error) {
+	totalCommunityRewards := big.NewInt(0)
+	// Fixed epoch reward to the infrastructure fund.
+	// TODO(joshua): The community reward needs to be split among different recipients
```
Comment (asaj) [inline consensus/istanbul/backend/pos.go]: Which recipients?

File: consensus/istanbul/backend/pos.go [hunk 4]
```diff
@@ -133,43 +141,47 @@ func (sb *Backend) updateValidatorScores(header *types.Header, state *state.Stat
...
-		state.AddBalance(*governanceAddress, communityEpochReward)
-		totalEpochRewards.Add(totalEpochRewards, communityEpochReward)
+		state.AddBalance(*governanceAddress, communityReward)
+		totalCommunityRewards.Add(totalCommunityRewards, communityReward)
 	}
 
+	return totalCommunityRewards, err
+}
+
+func (sb *Backend) distributeElectionRewards(header *types.Header, state *state.StateDB, valSet []istanbul.Validator, maxTotalRewards *big.Int, uptimes []*big.Int) (*big.Int, error) {
```
Comment (asaj) [inline consensus/istanbul/backend/pos.go]: Consider `distributeVoterRewards`

File: consensus/istanbul/backend/pos.go (hunk 3 above)
Comment (trianglesphere) [reply consensus/istanbul/backend/pos.go]: If the reserve is low, all to the reserve.
If the reserve if fine: 2% to the eco fund & 98% to the community fund is planned. I might send it all to the governance contract right now.

File: consensus/istanbul/backend/pos.go (hunk 2 above)
Comment (trianglesphere) [reply consensus/istanbul/backend/pos.go]: The recipients described below.

**Pull request merged**